helix-core/src/test/java/org/apache/helix/controller/stages/TestSendCancellationMessageFromOFFLINEToDROPPED.java

```diff
@@ -0,0 +1,97 @@
+package org.apache.helix.controller.stages;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.helix.HelixDefinedState;
+import org.apache.helix.HelixManager;
+import org.apache.helix.controller.common.PartitionStateMap;
+import org.apache.helix.controller.common.ResourcesStateMap;
+import org.apache.helix.controller.dataproviders.BaseControllerDataProvider;
+import org.apache.helix.model.ClusterConfig;
+import org.apache.helix.model.LiveInstance;
+import org.apache.helix.model.Message;
+import org.apache.helix.model.Partition;
+import org.apache.helix.model.Resource;
+import org.apache.helix.model.StateModelDefinition;
+import org.apache.helix.task.TaskConstants;
+import org.apache.helix.tools.StateModelConfigGenerator;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * This test checks the cancellation message generation when currentState=null and desiredState=DROPPED
+ */
+public class TestSendCancellationMessageFromOFFLINEToDROPPED extends MessageGenerationPhase {
```

Comment:
Can we just add this test method to some existing test classes? And the name is too long.

Reply:
I'll change it to a shorter name. It has to be a separate class because it needs to extend MessageGenerationPhase so I can run this class alone. Since it is a very specific case, it's better we can run this stage alone so we can control the input to this message generation phase.

Reply:
Sure, sounds good.